native_client/deepspeech.h

```diff
@@ -170,6 +189,20 @@ char* DS_IntermediateDecode(StreamingState* aSctx);
 DEEPSPEECH_EXPORT
 char* DS_FinishStream(StreamingState* aSctx);
 
+/**
+ * @brief Signal the end of an audio signal to an ongoing streaming
+ *        inference, returns the STT result over the whole audio signal plus metadata.
+ *
+ * @param aSctx A streaming state pointer returned by {@link DS_SetupStream()}.
+ *
+ * @return The STT result plus metadata on word timings as a JSON string. 
+ *		   The user is responsible for freeing the string.
+ *
+ * @note This method will free the state pointer (@p aSctx).
+ */
+DEEPSPEECH_EXPORT
+char* DS_FinishStreamExtended(StreamingState* aSctx);
```

Comment:
`metadata_from_output` depends on `ModelState::duration_secs` being set, which doesn't happen unless you call `DS_SpeechToTextExtended`, so there's no point in exposing this function in the API, it doesn't work independently.